download-kafka.sh

```diff
@@ -1,11 +1,12 @@
 #!/bin/sh -e
 
-mirror=$(curl --stderr /dev/null 'https://www.apache.org/dyn/closer.cgi?as_json=1' | jq -r '.preferred')
+path="kafka/${KAFKA_VERSION}/kafka_${SCALA_VERSION}-${KAFKA_VERSION}.tgz"
+mirror=$(curl --stderr /dev/null "https://www.apache.org/dyn/closer.cgi?path=/${path}&as_json=1" | jq -r '.preferred')
 
 if [[ -z "$mirror" ]]; then
 	echo "Unable to determine mirror for downloading Kafka, the service may be down"
 	exit 1
 fi
 
-url="${mirror}kafka/${KAFKA_VERSION}/kafka_${SCALA_VERSION}-${KAFKA_VERSION}.tgz"
+url=${mirror}${path}
```

Comment:
This looks a bit weird from a bash scripting perspective. Perhaps it's better to just use the response from `closer.cgi` to build the full URL, such as:

```
url=$(curl "https://www.apache.org/dyn/closer.cgi?path=${path}&as_json=1" | jq -r '"\(.preferred)\(.path_info)")'
```

Reply:
Thanks for review, I made changes in my pull request